src/iks003_incompressibleLinearElasticity.cpp

```diff
@@ -33,55 +33,37 @@
-struct Solid {
+template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false, bool useFlat = true>
+struct Solid : public FEBase<Basis_, useFlat, FERequirements_, useEigenRef> {
  public:
-  using Traits            = FETraits<Basis_, FERequirements_, useEigenRef>;
-  using Basis             = typename Traits::Basis;
+  using Base              = FEBase<Basis_, useFlat, FERequirements_, useEigenRef>;
+  using Traits            = typename Base::Traits;
+  using BasisHandler      = typename Traits::BasisHandler;
   using FlatBasis         = typename Traits::FlatBasis;
   using FERequirementType = typename Traits::FERequirementType;
   using LocalView         = typename Traits::LocalView;
   using Geometry          = typename Traits::Geometry;
   using Element           = typename Traits::Element;
+  using GlobalIndex       = typename Traits::GlobalIndex;
 
-  Solid(const Basis &basis, const typename LocalView::Element &element, double emod, double nu)
-      : localView_{basis.flat().localView()}, emod_{emod}, nu_{nu} {
-    localView_.bind(element);
+  Solid(const BasisHandler &basisHandler, const typename LocalView::Element &element, double emod, double nu)
+      : Base(basisHandler, element), emod_{emod}, nu_{nu} {
     mu_       = emod_ / (2 * (1 + nu_));
     lambdaMat = convertLameConstants({.emodul = emod_, .nu = nu_}).toLamesFirstParameter();
   }
 
-  using GlobalIndex = typename LocalView::MultiIndex;
-  void globalFlatIndices(std::vector<GlobalIndex> &globalIndices) const {
-    const auto &feDisp = localView_.tree().child(_0, 0).finiteElement();
-    for (size_t i = 0; i < feDisp.size(); ++i) {
-      for (int j = 0; j < Traits::worlddim; ++j) {
-        globalIndices.push_back(localView_.index((localView_.tree().child(_0, j).localIndex(i))));
-      }
-    }
-    const auto &fePressure = localView_.tree().child(_1).finiteElement();
-    for (size_t i = 0; i < fePressure.size(); ++i) {
-      globalIndices.push_back(localView_.index((localView_.tree().child(_1).localIndex(i))));
-    }
-  }
-
   inline double calculateScalar(const FERequirementType &par) const { return calculateScalarImpl<double>(par); }
 
-  [[nodiscard]] constexpr int size() const { return localView_.size(); }
-  const Element &gridElement() { return localView_.element(); }
-  const LocalView &localView() const { return localView_; }
-  LocalView &localView() { return localView_; }
-
  protected:
   template <class ScalarType>
   auto calculateScalarImpl(const FERequirementType &par, const std::optional<const Eigen::VectorX<ScalarType>> &dx
                                                          = std::nullopt) const -> ScalarType {
     const auto &d      = par.getGlobalSolution(Ikarus::FESolutions::displacement);
     const auto &lambda = par.getParameter(Ikarus::FEParameter::loadfactor);
-    Eigen::VectorX<ScalarType> localDisp(localView_.size());
+    Eigen::VectorX<ScalarType> localDisp(this->localView().size());
```

Comment:
by anticipating that we want to use on the tree in the future, you can refactor this to 

```cpp
  const auto& tree = this->localView().tree();
  Eigen::VectorX<ScalarType> localDisp(tree .size());
    localDisp.setZero();
    auto &displacementNode = tree .child(_0, 0);
    auto &pressureNode     = tree .child(_1);
```

Reply:
This is a nice idea, thanks. I think storing `tree` as a variable will simplify things in the next four instances and it will not affect the current documentation also. I will do this change after our discuss for the other comment.

In general, as we wanted to keep the PR short and particular to specific purposes, I think refactoring based on possible future changes should not be done as then converging to a particular state can take a longer time.

Reply:
Is this modification of storing `tree` expected also for `iks004` and `iks007`? For `iks007` may be one could also use the function `localSolutionBlockVector`, but may be these refactoring were not within the scope of this PR?